node-js-basics-101/node_modules/uws/uws.js

```diff
@@ -0,0 +1,563 @@
+
+        if (!options.noServer) {
+            this.httpServer = options.server ? options.server : http.createServer((request, response) => {
+                // todo: default HTTP response
+                response.end();
+            });
+
+            if (options.path && (!options.path.length || options.path[0] !== '/')) {
+                options.path = '/' + options.path;
+            }
+
+            this.httpServer.on('upgrade', this._upgradeListener = ((request, socket, head) => {
+                if (!options.path || options.path == request.url.split('?')[0].split('#')[0]) {
+                    if (options.verifyClient) {
+                        const info = {
+                            origin: request.headers.origin,
+                            secure: request.connection.authorized !== undefined || request.connection.encrypted !== undefined,
+                            req: request
+                        };
+
+                        if (options.verifyClient.length === 2) {
+                            options.verifyClient(info, (result, code, name) => {
+                                if (result) {
+                                    this.handleUpgrade(request, socket, head, emitConnection);
+                                } else {
+                                    abortConnection(socket, code, name);
+                                }
+                            });
+                        } else {
+                            if (options.verifyClient(info)) {
+                                this.handleUpgrade(request, socket, head, emitConnection);
+                            } else {
+                                abortConnection(socket, 400, 'Client verification failed');
+                            }
+                        }
+                    } else {
+                        this.handleUpgrade(request, socket, head, emitConnection);
+                    }
+                } else {
+                    if (this._lastUpgradeListener) {
+                        abortConnection(socket, 400, 'URL not supported');
+                    }
+                }
+            }));
+
+            this.httpServer.on('newListener', (eventName, listener) => {
+                if (eventName === 'upgrade') {
+                    this._lastUpgradeListener = false;
+                }
+            });
+
+            this.httpServer.on('error', (err) => {
+                this.emit('error', err);
+            });
+        }
+
+        native.server.group.onDisconnection(this.serverGroup, (external, code, message, webSocket) => {
+            webSocket.external = null;
+
+            process.nextTick(() => {
+                webSocket.internalOnClose(code, message);
+            });
+
+            native.clearUserData(external);
+        });
+
+        native.server.group.onMessage(this.serverGroup, onServerMessage);
+
+        native.server.group.onPing(this.serverGroup, (message, webSocket) => {
```

Comment:
__[jshint]__
* 'arrow function syntax (=>)' is only available in ES6 (use 'esversion: 6').

<!-- Comment by [SideCI](https://sideci.com) -->